### ISSUE TITLE
The non-abstract class 'HtmlFlex' is missing implementations for these members

### ISSUE THREAD

**Issue by mark8044** [daohoangson/flutter_widget_from_html#1312]
I started to get this error after upgrade to the latest master branch version, I haven't don't an update for about a week and a half, so I'm not sure when this appeared exactly:

```
Error (Xcode): ../../../.pub-cache/hosted/pub.dev/flutter_widget_from_html_core-0.15.1/lib/src/widgets/html_flex.dart:38:7: Error: The non-abstract class 'HtmlFlex' is missing implementations for these members:
```

Oddly, it doesn't actually go on to tell me the members...

this is on the latest master channel version:
```
Flutter 3.24.0-1.0.pre.528 • channel master • https://github.com/flutter/flutter.git
Framework • revision 493c453d57 (4 hours ago) • 2024-08-09 12:53:15 +0200
Engine • revision c9ec468c75
Tools • Dart 3.6.0 (build 3.6.0-131.0.dev) • DevTools 2.38.0
```

**Comment by mrvyas417** [comment]
Same issue 

class _HtmlFlexRenderObject extends RenderBox
      ^^^^^^^^^^^^^^^^^^^^^
: Context: 'RenderFlex.spacing' is defined here.
flex.dart:647
  double get spacing => _spacing;
             ^^^^^^^
: Context: 'RenderFlex.spacing=' is defined here.
flex.dart:649
  set spacing (double value) {
      ^^^^^^^

: Error: The method 'hashValues' isn't defined for the class 'SlidableRatioNotification'.
notifications_old.dart:88
- 'SlidableRatioNotification' is from 'package:flutter_slidable/src/notifications_old.dart' ('../.pub-cache/hosted/pub.dev/flutter_slidable-3.1.1/lib/src/notifications_old.dart').
notifications_old.dart:1
Try correcting the name to the name of an existing method, or defining a method named 'hashValues'.
  int get hashCode => hashValues(tag, ratio);
                      ^^^^^^^^^^

**Comment by mark8044** [comment]
> Same issue
> 
> class _HtmlFlexRenderObject extends RenderBox ^^^^^^^^^^^^^^^^^^^^^ : Context: 'RenderFlex.spacing' is defined here. flex.dart:647 double get spacing => _spacing; ^^^^^^^ : Context: 'RenderFlex.spacing=' is defined here. flex.dart:649 set spacing (double value) { ^^^^^^^
> 
> : Error: The method 'hashValues' isn't defined for the class 'SlidableRatioNotification'. notifications_old.dart:88
> 
> * 'SlidableRatioNotification' is from 'package:flutter_slidable/src/notifications_old.dart' ('../.pub-cache/hosted/pub.dev/flutter_slidable-3.1.1/lib/src/notifications_old.dart').
>   notifications_old.dart:1
>   Try correcting the name to the name of an existing method, or defining a method named 'hashValues'.
>   int get hashCode => hashValues(tag, ratio);
>   ^^^^^^^^^^

The second part of your error has to do with the slidable plugin, which I also had a problem with.  But the fix is currently on their master branch, just not pushed to pub.dev https://github.com/letsar/flutter_slidable/pull/484

**Comment by CheeseLad** [comment]
Having this issue also when trying to compile a web app

Compiling lib/main.dart for the Web...                          
Target dart2js failed: ProcessException: Process exited abnormally with exit code 1:
../.pub-cache/hosted/pub.dev/flutter_widget_from_html_core-0.15.1/lib/src/widgets/html_flex.dart:38:7:
Error: The non-abstract class 'HtmlFlex' is missing implementations for these members:
 - Flex.spacing
class HtmlFlex extends MultiChildRenderObjectWidget
      ^^^^^^^^
flutter/packages/flutter/lib/src/widgets/basic.dart:4715:16:
Info: 'Flex.spacing' is defined here.
  final double spacing;
               ^^^^^^^
../.pub-cache/hosted/pub.dev/flutter_widget_from_html_core-0.15.1/lib/src/widgets/html_flex.dart:175:7:
Error: The non-abstract class '_HtmlFlexRenderObject' is missing implementations for these members:
 - RenderFlex.spacing
 - RenderFlex.spacing=
class _HtmlFlexRenderObject extends RenderBox

**Comment by bekirtaskin** [comment]
Facing this as well.

**Comment by daohoangson** [comment]
Yes, there is a conflict between this package and Flutter master. I'm working on a fix.

**Comment by mark8044** [comment]
Is there a quick and dirty fix we can do in the meantime to get it working again?

**Comment by mark8044** [comment]
EDIT2, Im getting a ton of crashes with the above code.  Not sure if its a problem with the flutter framework or my bad code.  Use with caution

**Comment by daohoangson** [comment]
The proper fix will require Flutter 3.24 so I will do a workaround for v0.15 and release it today. The fix will be released as v0.16 since it is a breaking change.

**Comment by daohoangson** [comment]
`v0.15.2` has been released with the workaround. Please try upgrading and see whether it works with Flutter master. Thank you for your patience.

**Comment by mark8044** [comment]
@daohoangson 
It seems to be working with latest master branch 
```
Flutter 3.25.0-1.0.pre.91 • channel master • https://github.com/flutter/flutter.git
Framework • revision a9e94d9045 (11 minutes ago) • 2024-08-22 09:12:54 -0700
Engine • revision 67a7fe18c0
Tools • Dart 3.6.0 (build 3.6.0-167.0.dev) • DevTools 2.39.0-dev.15
```

I am getting all sorts of weird glitching on Android when I use HTMLWidget.  I don't know if its an issue with this repo or the underlying Flutter SDK, I suspect its a flutter issue as I see this similar problem reported: https://github.com/flutter/flutter/issues/153762

Otherwise thanks much for the fix!

**Comment by mark8044** [comment]
Ignore my previous comments, it looks like this is a bug with impeller and not to do with this repo 👍

**Comment by mamimotu** [comment]
Someone solved it???

**Comment by orenagiv** [comment]
In my case, I'm using the package [better_player_plus](https://github.com/SunnatilloShavkatov/better_player_plus) v1.0.7, which depends on `flutter_widget_from_html_code` v0.14.12.
So I've updated the `pubspec.yaml` to override the dependency with the latest `flutter_widget_from_html_code` (v0.15.2), and it seems to build successfully and work now:
```yaml
dependency_overrides:
  flutter_widget_from_html_core: ^0.15.2
```

**Comment by AbhiralJain** [comment]
@orenagiv This worked for me as well! Thank you for your comment!

**Comment by o7planning** [comment]
I had the same problem. And I fixed it.
This issue occurs when the Flutter version on the environment variable is less than 3.24.0, while the Flutter version declared in the Settings of Android Studio is the latest.
You need to change the path for the environment variable. Then check the "flutter --version" command on CMD to make sure it is the latest.

**Comment by hemendra-shaktawat** [comment]
@orenagiv This worked for me as well! Thank you for your comment!
